services/galley/src/Galley/API/Update.hs

```diff
@@ -814,6 +811,57 @@ postNewOtrMessage protectee con cnv val msg = do
       gone <- External.deliver toBots
       mapM_ (deleteBot cnv . botMemId) gone
 
+-- | Locally post a message originating from a remote conversation
+-- FUTUREWORK: error handling for missing / mismatched clients
+postRemoteToLocal ::
+  UTCTime ->
+  Remote ConvId ->
+  Qualified UserId ->
+  ClientId ->
+  Maybe Text ->
+  Maybe Priority ->
+  Bool ->
+  UserClientMap Text ->
```

Comment:
I think inventing a new internal datatype would be nicer than this long list of parameters. Why not pass RemoteMessage here?

Reply:
Yes, good idea. Done now.